pom.xml

```diff
@@ -495,6 +506,97 @@
+                  var artifactId = project.getProperty("project.artifactId");
+                  var version = project.getProperty("project.version");
+
+                  var cfgFile = new File(targetDir, artifactId + "-" + version + ".json");
+                  if (!cfgFile.exists()) {
+                    cfgFile.createNewFile();
+                  }
+
+                  var etlRange = project.getProperty("etl.versionRange");
+                  var config = {
+                    "parents": [ "system:cdap-etl-batch" + etlRange, "system:cdap-etl-realtime" + etlRange ],
+                    "properties": {}
+                  }
+
+                  // look in widgets directory for widget config for each plugin
+                  var widgetsDir = new File(baseDir, project.getProperty("widgets.dir"));
+                  if (widgetsDir.isDirectory()) {
+                    var widgetsFiles = widgetsDir.listFiles();
+                    for (i = 0; i < widgetsFiles.length; i++) {
+                      var widgetsFile = widgetsFiles[i];
+                      if (widgetsFile.isFile()) {
+                        var contents = new java.lang.String(Files.readAllBytes(widgetsFile.toPath()), StandardCharsets.UTF_8);
+                        var contentsAsJson = JSON.parse(contents);
+                        var propertyName = "widgets." + widgetsFile.getName();
+                        // if the filename ends with .json, strip the .json
+                        if (propertyName.indexOf(".json", propertyName.length - 5) !== -1) {
+                          propertyName = propertyName.slice(0, -5);
+                        }
+                        config.properties[propertyName] = JSON.stringify(contentsAsJson);
+                      }
+                    }
+                  }
+
+                  // look in the docs directory for docs for each plugin
+                  var docsDir = new File(baseDir, project.getProperty("docs.dir"));
+                  if (docsDir.isDirectory()) {
+                    var docFiles = docsDir.listFiles();
+                    for (i = 0; i < docFiles.length; i++) {
+                      var docFile = docFiles[i];
+                      if (docFile.isFile()) {
+                        var contents = new java.lang.String(Files.readAllBytes(docFile.toPath()), StandardCharsets.UTF_8);
+                        var propertyName = "doc." + docFile.getName();
+                        // if the filename ends with .md or .html, strip the extension
+                        if (propertyName.indexOf(".md", propertyName.length - 3) !== -1) {
+                          propertyName = propertyName.slice(0, -3);
+                        }
+                        if (propertyName.indexOf(".html", propertyName.length - 5) !== -1) {
+                          propertyName = propertyName.slice(0, -5);
+                        }
+                        config.properties[propertyName] = contentsAsJson;
```

Comment:
if there are two files, ``Readme.md and  Readme.html`` in the same dir, then one of them will be set to config.properties, replacing the other.  is that okay?